core/app/models/spree/product/scopes.rb

```diff
@@ -62,7 +62,7 @@ def self.property_conditions(property)
     #   SELECT COUNT(*) ...
     add_search_scope :in_taxon do |taxon|
       includes(:classifications).
-      where("spree_products_taxons.taxon_id" => taxon.self_and_descendants.pluck(:id)).
+      where("spree_classifications.taxon_id" => taxon.self_and_descendants.pluck(:id)).
```

Comment:
Wthis could be also written as:

~~~ RUBY
where(spree_classifications: { taxon_id: taxon.self_and_descendats.pluck(:id) })
~~~